### PR TITLE
Send async messages to the Android main looper

### LINKED ISSUE #427: Make Android UI dispatcher asynchronous by default
See discussion in #381 

We can make `UI` asynchronous by default and provide a fallback (system property?) to sync behaviour and something like `SyncUI`.

Asynchronous handlers are the new feature in Android and it's unclear for us (as we're not Android devs) whether it's sane default which is safe to expose, so we'd like to hear community opinion on this.

### PR DESCRIPTION
**Pull request by JakeWharton**
Closes #427

### CODE REVIEW

**Comment by LouisCAD** [comment]
Is this a problem regarding the SDK used to compile it?

**Comment by JakeWharton** [comment]
Yes as it does not use AGP

**Comment by qwwdfsad** [comment]
Rebased onto develop.
Thank you!

File: ui/kotlinx-coroutines-android/src/HandlerDispatcher.kt [hunk 1]
```diff
@@ -40,7 +42,32 @@ public fun Handler.asCoroutineDispatcher(): HandlerDispatcher =
 
 private const val MAX_DELAY = Long.MAX_VALUE / 2 // cannot delay for too long on Android
 
-private val mainHandler = Handler(Looper.getMainLooper())
+private val mainHandler = Looper.getMainLooper().asHandler(async = true)
```
[the remark this answers is not in the document]
Comment (qwwdfsad) [reply ui/kotlinx-coroutines-android/src/HandlerDispatcher.kt]: Good point.
Let's wait for the feedback from the users and will discuss solutions for their particular problems.

This PR carries enough value to release it as is.

**Pull request closed without merging**